RiiR

isabelroses opened the pull request:
@uncenter :p

isabelroses committed:
feat!: delete the old

isabelroses committed:
init v2

isabelroses committed:
feat: we're cooking

isabelroses committed:
ci: add testing via nix

uncenter commented on README.md:
```diff
@@ -5,28 +5,23 @@
   Blåhaj
 </h1>
 
```
> It is written in JavaScript and uses the discord.js library.

mfw

isabelroses replied:
anioop

uncenter commented on README.md:
````diff
@@ -5,28 +5,23 @@
 To run blahaj
 ```bash
-node src/index.js
+go run .
````
This shit is not written in Go 😭

isabelroses replied:
UMMMMMMMMMMMMM

uncenter commented on src/commands/avatar.rs:
```diff
@@ -0,0 +1,15 @@
+    #[description = "Selected user"] user: Option<serenity::User>,
+) -> Result<()> {
+    let u = user.as_ref().unwrap_or_else(|| ctx.author());
+    let av = u.avatar_url().expect("Could not get avatar URL");
```
Why are you obfuscating variable names this is not C

isabelroses replied:
go habbits tbh

uncenter commented on src/commands/bot.rs:
```diff
@@ -0,0 +1,24 @@
+/// Displays information about the bot
+#[poise::command(slash_command)]
+pub async fn botinfo(ctx: Context<'_>) -> Result<()> {
+    let rev = env::var("BUILD_REV").unwrap_or_else(|_| "unknown".to_string());
```
Pretty sure you don't need a closure here if you just do something like this
```suggestion
    let rev = env::var("BUILD_REV").unwrap_or("unknown".to_string());
```

isabelroses replied:
:o

uncenter commented on src/main.rs:
```diff
@@ -0,0 +1,75 @@
+async fn main() -> Result<()> {
+    env_logger::init();
+
+    // Load the .env file
+    dotenv().ok();
+
+    // Enable color_eyre beacuse error handling ig
+    color_eyre::install()?;
```
Least helpful comments ive ever seen lol

isabelroses replied:
it helps me •`ᴖ´•

uncenter replied:
What else could color_eyre::install do besides install color_eyre and what else could dotenv() be doing besides loading the dot env file 😭

isabelroses committed:
docs: yap

isabelroses committed:
feat!: no logging

isabelroses committed:
style: less confusing names

isabelroses committed:
what would you even call this

uncenter commented on README.md:
```diff
@@ -1,26 +1,13 @@
 </h1>
 
-Blahaj is a discord bot that is designed to do anything that you want it to do. It is written in JavaScript and uses the discord.js library.
+Blahaj a blazzingly fasy discord bot written in rust. Rocketship emoji.
```
You spelled "blazingly" wrong LMAO

isabelroses replied:
its joever

uncenter commented on README.md:
````diff
@@ -1,26 +1,13 @@
-```bash
-go run .
-```
+I don't know use nix?
````
mfw

isabelroses replied:
winnable

uncenter commented on src/commands/avatar.rs:
good job

isabelroses replied:
ty lil sis

uncenter commented on src/commands/bot.rs:
call it better code

isabelroses replied:
:p

isabelroses committed:
docs: typo

isabelroses committed:
chore: bump deps

isabelroses committed:
feat: kittysay

awesome way to abitarily inject my computor with a command to remove all
my files

isabelroses committed:
feat: whois command

isabelroses committed:
feat: code expantion

isabelroses committed:
refactor: shenanians in prep for code expantion with gitea/forgejo

uncenter commented on src/event_handler/code_expantion.rs:
```diff
@@ -0,0 +1,109 @@
+        if re.captures(content.as_str()).is_some() {
+            let mut code_blocks = Vec::new();
+
+            for caps in re.captures_iter(content) {
```
Why execute the regex twice

isabelroses replied:
um ops

isabelroses replied:
really good point tbh

uncenter replied:
lmao yw

uncenter commented on src/event_handler/code_expantion.rs:
```diff
@@ -0,0 +1,109 @@
+                    end_str.as_str().parse::<usize>().unwrap_or(start)
+                } else {
+                    start
+                };
```
this is cursed

isabelroses replied:
tell me about it

isabelroses replied:
working on some chef work, cursed regex magic for gitea/forgejo instance hopefully looks a bit cleaner

uncenter commented on src/event_handler/code_expantion.rs:
```diff
@@ -0,0 +1,109 @@
+                        if end > start {
+                            format!("-{}", end)
+                        } else {
+                            "".to_string()
```
Use `String::new()`

uncenter replied:
In general you should be applying clippy suggestions, add this to your Cargo.toml:
```toml
[lints.clippy]
all = "warn"
pedantic = "warn"
```

isabelroses replied:
:o thats pretty cool

uncenter replied:
yeah and run `cargo clippy --fix` i think

isabelroses committed:
fix: nix unknown rev

isabelroses committed:
sort deps

isabelroses committed:
feat: git instance compatable with github source tree thing ig

isabelroses committed:
chore(nix): add extra deps

isabelroses committed:
fix(nix): url

isabelroses committed:
chore: apply various code suggestions

isabelroses committed:
ci: run on prs

comfysage committed:
feat: roll command (#2)

Co-authored-by: Isabel <isabel@isabelroses.com>
